x/httpproxy/proxy_handler.go

```diff
@@ -43,6 +53,7 @@ func (h *proxyHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.R
 func NewProxyHandler(dialer transport.StreamDialer) http.Handler {
 	return &proxyHandler{
 		connectHandler: NewConnectHandler(dialer),
+		pathHandler:    NewPathHandler(dialer),
 		forwardHandler: NewForwardHandler(dialer),
```

Comment:
Perhaps we should bypass the public constructors so we can share the client between the path and forward handlers.